File: Benchmark/Rust/Compute/Pi.rs
fn calculate_pi(terms: usize) -> f64 {
    let mut pi = 0.0;
    for k in 0..terms {
        if k % 2 == 0 {
            pi += 1.0 / (2 * k + 1) as f64;
        } else {
            pi -= 1.0 / (2 * k + 1) as f64;
        }
    }
    4.0 * pi
}

fn main() {
    let terms = 1_000_000;
    let pi = calculate_pi(terms);
    println!("Estimated Pi: {}", pi);
}

File: Benchmark/Compute/Pi/Pi.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func calculatePi(terms int) float64 {
    pi := 0.0
    for k := 0; k < terms; k++ {
        if k%2 == 0 {
            pi += 1.0 / float64(2*k+1)
        } else {
            pi -= 1.0 / float64(2*k+1)
        }
    }
    return 4 * pi
}

func main() {
    terms, _ := strconv.Atoi(os.Args[1])
    pi := calculatePi(terms)
    fmt.Printf("%.12f\n", pi)
}